### ISSUE TITLE
CarFusion generated by your guidance is not consistent with your annotations.

### ISSUE THREAD

**Issue by kltrock** [luminxu/Pose-for-Everything#3]
I followed your guidance to generate CarFusion. 

First, I downloaded all the zip files from CarFusion website and extract them into:
-train: car_butler1  car_butler2  car_craig1  car_craig2  car_fifth1  car_fifth2  car_morewood1  car_morewood2
-test: car_penn1  car_penn2

Second, I used the [official tools](https://github.com/dineshreddy91/carfusion_to_coco) to convert the annotations to COCO format, resulting in car_keypoints_test.json, car_keypoints_train.json.

Finally, I renamed all the jpg files using your [rename_carfusion_image.py](https://github.com/luminxu/Pose-for-Everything/blob/main/mp100/rename_carfusion_image.py).

However, the images obtained are not consistent with your annotation files. For example, in your annotation files, there is an image with name 'car/1500801201.jpg' which is missing in the generated images.

**Comment by kltrock** [comment]
Could you please provide more details on how to obtain the CarFusion correctly?

**Comment by luminxu** [comment]
Thanks for your question. I am wondering whether all the annotations are inconsistent with the generated images or only some of them are missing. Would you mind providing some examples of the names of the generated images?

**Comment by kltrock** [comment]
I think all the annotations are inconsistent. As I have described earlier, I downloaded the Carfusion datasets, which resulted in 10 folders (car_butler1, car_butler2, car_craig1, car_craig2, car_fifth1, car_fifth2, car_morewood1, car_morewood2, car_penn1, car_penn2).

However, following the [official tools](https://github.com/dineshreddy91/carfusion_to_coco/blob/master/carfusion2coco.py#L164), the image_id of each image is defined by 'loop', 'video_id', 'frame_id'. The 'loop' variable is loosely defined by the order of the 10 folders. The highest value of the 'loop' variable is 10, so the image_id would be 10xxxxxxxx.jpg, 9xxxxxxxx.jpg, etc. However, in your annotations, there are a lot of annotations with image names 15xxxxxxxx.jpg, 14xxxxxxxx.jpg, etc.

![Screen Shot 2022-09-12 at 15 07 30](https://user-images.githubusercontent.com/109402545/189593151-245d8f09-0c16-416c-a213-63c6bf63f414.png)

Could you please provide more details or a script to correctly generate the dataset?

**Comment by A465539338** [comment]
I found the same problem as kltrock. And there is another question for me. It seems all the train and test images from CarFusion dataset shares the only category car which means no annotation for bus and suv.  Could you please tell more about how to arrage this CarFusion dataset?

**Comment by Annbless** [comment]
> Thanks for your question. I am wondering whether all the annotations are inconsistent with the generated images or only some of them are missing. Would you mind providing some examples of the names of the generated images?

It seems that all the annotations are inconsistent with the generated images. I have noticed that the names of the images generated following the guidance have 9-digit numbers while the names provided in the annotations have 10-digit numbers.

**Comment by worklikerlei** [comment]
> 



> I found the same problem as kltrock. And there is another question for me. It seems all the train and test images from CarFusion dataset shares the only category car which means no annotation for bus and suv. Could you please tell more about how to arrage this CarFusion dataset?

Hi, I have also encountered this problem. Have you solved it?

**Comment by worklikerlei** [comment]
Hello, have you solved the problem of this dataset？

**Comment by kltrock** [comment]
Yes. I check every instance in their annotation files and find the corresponding image accordingly.

**Comment by worklikerlei** [comment]
> Yes. I check every instance in their annotation files and find the corresponding image accordingly.

Thank you very much for your reply. following the https://github.com/dineshreddy91/carfusion_to_coco/blob/master/carfusion2coco.py#L164  
It seems all the train and test images from CarFusion dataset shares the only category car which means no annotation for bus and suv. Could you give me some help. Thank you very much!

**Comment by worklikerlei** [comment]
Hello, for the mp100 dataset in the Pose-for-Everything task, the carfusion datasets image does not match. Can you provide some ideas to help me? Thank you very much.









--

| |
fengzaifei12
|
|
***@***.***
|



在 2023-02-02 16:48:26，"kltrock" ***@***.***> 写道：

Yes. I check every instance in their annotation files and find the corresponding image accordingly.

**Comment by kltrock** [comment]
I suggest you examine every instance in the annotation files of Carfusion and Mp100. You can match one instance in Carfusion with one instance in Mp100 if they have identical information (keypoint positions, number of key points, etc). After matching, you can trace back the images in the Carfusion.

**Comment by worklikerlei** [comment]
There are more than 30000 pictures generated by carfusion_ to_ coco. It is very difficult for each annotaiton to find pictures. What are the skills about this?









--

| |
fengzaifei12
|
|
***@***.***
|



在 2023-02-09 15:54:48，"kltrock" ***@***.***> 写道：

I suggest you examine every instance in the annotation files of Carfusion and Mp100. You can match one instance in Carfusion with one instance in Mp100 if they have identical information (keypoint positions, number of key points, etc). After matching, you can trace back the images in the Carfusion.

**Comment by kltrock** [comment]
You can write a script for it. It would be fast.

**Comment by rika1024** [comment]
I also encountered this problem, have you solved it?

**Comment by rika1024** [comment]
I also encountered this problem, can you share your script? Thank you

**Comment by festla** [comment]
I have encountered the same problem with you. And  I wonder if I need to process all the 10 files or only one is sufficient.
I have done the convert_to_coco and the rename,but my situation is kind of this->

![Image](https://github.com/user-attachments/assets/f9cb9ad3-4b6e-4b82-aecc-e92f535420a1)
